packages/lit-dev-content/site/articles/article/lit-cheat-sheet.md

```diff
@@ -0,0 +1,1118 @@
+but the ones you would normally use are the following and this is their general
+order:
+
+1. `constructor` – (Native custom element lifecycle)
+2. `connectedCallback` – (Native)
+3. `willUpdate` – (Lit lifecycle)
+4. `update` – (Lit)
+5. `render` – (Lit)
+6. `firstUpdated` – (Lit)
+7. `updated` – (Lit)
+8. `disconnectedCallback` – (Native)
+
+{% aside "warn" %}
+
+The Lit lifecycle and the native custom element lifecycle are distinct and managed separately.
+
+This means that while they generally follow a specific order, they may intermix
+because the browser controls the native lifecycle, while Lit and JavaScript
+manage the Lit lifecycle.
+
+For example, a component may be attached to the DOM and then removed before the
+Lit lifecycle may run at all, or a component may be created with
+`document.createElement` which would call the `constructor`, but if it's never
+added to the DOM, the `connectedCallback` would never run and thus the Lit
+lifecycle would never run.
+
+{% endaside %}
+
+**Related Documentation & Topics:**
+
+- [Lifecycle](/docs/components/lifecycle/)
+- [Lifecycle diagram](/docs/components/lifecycle/#reactive-update-cycle)
+
+#### `constructor`
+
+- Runs when the element is created via:
+  - `document.createElement('my-element')`
+  - `element.innerHTML = '<my-element></my-element>'`
+  - `new MyElement()`
+  - etc.
+- A good place to set initial properties
+- Do **NOT** require parameters in the constructor or modify DOM
+
+#### `connectedCallback`
+
+- *May* run on the server. This has not been finalized.
+- Runs when the element is added to the DOM via:
+  - `element.appendChild(element)`
+  - `element.innerHTML = '<my-element></my-element>'`
+  - etc.
+- Can run multiple times, but a good place to set up event listeners
+
+#### `willUpdate`
+
+- Runs on the server – do **NOT** access the DOM or browser APIs
+- A good place to set up properties that depend on other properties
+
+#### `update`
+
+- A good place to update properties that depend on other properties that depend on DOM
+
+#### `render`
+
+- Runs on the server - do NOT access the DOM or browser APIs
+
+#### `firstUpdated`
+
+- Runs after the first render
+- A good place to perform initializations that require access to the component's
+  rendered DOM
+
+#### `updated`
+
+- A good place to do some updates that require the component's rendered DOM or
+  to update properties that depend on the rendered DOM
+- Avoid setting reactive properties in this lifecycle as doing so may trigger
+  unnecessary re-renders. Try to do them in `willUpdate` or `update` if
+  possible.
+
+#### `disconnectedCallback`
+
+- Is called *AFTER* the element is removed from the DOM
+- A good place to clean up event listeners
+
+### Waiting for an update
+
+All Lit elements have asynchronous lifecycles. You need to wait for the `updateComplete` promise to resolve before you can be sure that the element has finished updating its DOM.
```

Comment:
```suggestion
All Lit elements have asynchronous lifecycles. The reason for this is so that property changes (e.g. el.foo = 1; el.bar = 2;) are batched for efficiency and correctness. 

You need to wait for the `updateComplete` promise to resolve before you can be sure that the element has finished updating its DOM.
```